extensions/wikia/RobotsTxt/tests/WikiaRobotsTest.php

```diff
@@ -123,37 +123,39 @@ public function dataProviderNonProductionEnvironment() {
 	}
 
 	/**
-	 * Test Wikia\RobotsTxt\WikiaRobots sets the proper sitemap based on wgServer
+	 * Test Wikia\RobotsTxt\WikiaRobots sets the proper sitemap based on wgServer,
+	 * wgEnableSpecialSitemapExt, wgEnableSitemapXmlExt and wgSitemapXmlExposeInRobots
+	 *
+	 * @dataProvider dataProviderSitemap
 	 */
-	public function testSitemapEnabled() {
+	public function testSitemap( $wgEnableSpecialSitemapExt, $wgEnableSitemapXmlExt, $wgSitemapXmlExposeInRobots, $sitemapUrls ) {
 		$this->mockGlobalVariable( 'wgWikiaEnvironment', WIKIA_ENV_PROD );
 		$this->mockGlobalVariable( 'wgServer', 'http://server' );
-		$this->mockGlobalVariable( 'wgEnableSpecialSitemapExt', true );
+		$this->mockGlobalVariable( 'wgEnableSpecialSitemapExt', $wgEnableSpecialSitemapExt );
+		$this->mockGlobalVariable( 'wgEnableSitemapXmlExt', $wgEnableSitemapXmlExt );
+		$this->mockGlobalVariable( 'wgSitemapXmlExposeInRobots', $wgSitemapXmlExposeInRobots );
 
 		$robotsTxtMock = $this->getRobotsTxtMock();
 		$pathBuilderMock = $this->getPathBuilderMock();
 
 		$wikiaRobots = new WikiaRobots( $pathBuilderMock );
 		$wikiaRobots->configureRobotsBuilder( $robotsTxtMock );
 
-		$this->assertEquals( $robotsTxtMock->spiedSitemap, [ 'http://server/sitemap-index.xml' ] );
+		$this->assertEquals( $robotsTxtMock->spiedSitemap, $sitemapUrls );
 	}
 
-	/**
-	 * Test Wikia\RobotsTxt\WikiaRobots doesn't set the sitemap if wgEnableSpecialSitemapExt is false
-	 */
-	public function testSitemapDisabled() {
-		$this->mockGlobalVariable( 'wgWikiaEnvironment', WIKIA_ENV_PROD );
-		$this->mockGlobalVariable( 'wgServer', 'http://server' );
-		$this->mockGlobalVariable( 'wgEnableSpecialSitemapExt', false );
-
-		$robotsTxtMock = $this->getRobotsTxtMock();
-		$pathBuilderMock = $this->getPathBuilderMock();
-
-		$wikiaRobots = new WikiaRobots( $pathBuilderMock );
-
-		$wikiaRobots->configureRobotsBuilder( $robotsTxtMock );
-		$this->assertEquals( $robotsTxtMock->spiedSitemap, [] );
+	public function dataProviderSitemap() {
+		return [
+			# $wgEnableSpecialSitemapExt, $wgEnableSitemapXmlExt, $wgSitemapXmlExposeInRobots, $sitemapUrls
+			[ false, false, false, [] ],
+			[ false, false, true, [] ],
+			[ false, true, false, [] ],
+			[ false, true, true, [ 'http://server/sitemap-newsitemapxml-index.xml' ] ],
+			[ true, false, false, [ 'http://server/sitemap-index.xml' ] ],
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "http://server/sitemap-index.xml" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)